src/components/Verify.tsx

```diff
@@ -1,87 +1,79 @@
-  const [creationTxHashes, setCreationTxHashes] = useState<CreationTxHash[]>([
-    { chain: SUPPORTED_CHAINS.mainnet, hash: undefined },
-  ]);
-  const handleChainChange = (index: number, chain: Chain) => {
-    handleTxHashInput(index, 'chain', chain);
-    setChain(chain);
-  };
-
-  const handleTxHashInput = (index: number, kind: 'chain' | 'hash', value: string | Chain) => {
-    if (kind === 'chain') {
-      const chain = value as Chain;
-      setCreationTxHashes((prev) => {
-        const next = [...prev];
-        next[index] = { ...next[index], chain };
-        return next;
-      });
-      console.log('creationTxHashes1:', creationTxHashes);
-    } else if (kind === 'hash') {
-      const hash = value as unknown as Hash;
-      setCreationTxHashes((prev) => {
-        const next = [...prev];
-        next[index] = { ...next[index], hash };
-        return next;
-      });
-      console.log('creationTxHashes2:', creationTxHashes);
-    }
-  };
-
-  const handleInput = (e: React.FormEvent<HTMLInputElement>) => {
-    if (!(e.target instanceof HTMLInputElement)) return;
-    const key = e.target.id;
-    if (['repoUrl', 'repoCommit', 'contractAddress'].includes(key)) {
-      setFormData({ ...formData, [key]: e.target.value });
-    } else if (key === 'buildHint') {
-      setFormData({
-        ...formData,
-        buildConfig: { ...formData.buildConfig, buildHint: e.target.value },
-      });
-    } else {
-      throw new Error(`Unexpected input ID: ${key}`);
-    }
-    console.log('formData:', formData);
-  };
+  const onSubmit = handleSubmit(async (values) => {
+    console.log('formData', values);
+  });
 
   return (
     <>
       <div className='flex min-h-full flex-1 flex-col justify-center px-6 py-12 lg:px-8'>
         <div className='mt-10 sm:mx-auto sm:w-full sm:max-w-sm'>
-          <form action='#' method='POST'>
+          <form action='#' method='POST' onSubmit={onSubmit}>
             {/* Repo URL */}
             <label htmlFor='repoUrl' className='block text-sm font-medium leading-6 text-gray-900'>
-              Repo URL
+              Repo URL<span className='text-red-600'> *</span>
             </label>
             <div>
               <input
                 id='repoUrl'
-                onBlur={handleInput}
-                required
-                className='block w-full rounded-md border-0 py-1.5 pl-2 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6'
+                className='block w-full rounded-md border-0 py-1.5 pl-2 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-900 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6'
+                {...register('repoUrl', {
+                  required: 'This is required',
+                  pattern: {
+                    value: /^http(s?)(:\/\/)((www.)?)([a-zA-z0-9\-_]+)(.*)$/i,
```

Comment:
I didn't know how strict we wanted the validation so feel free to adjust and change. There should be an example of all the different types of validation. I also made all the fields required.

Reply:
Just tested the backend and it will error if the `https` part isn't present, so this validation seems right and we should make sure the error message says something like "Please enter a valid URL that starts with 'https'" or something like that. 

Where is this regex from? The one I've used in the passed is https://regex101.com/library/wP2nP6, which we can modify to make the https part mandatory, but if this is from stackoverflow or something we should include a link